Strip package root part of filename before encoding the filename for …

4kochi opened the pull request:
…the cache

Closes #158

kulshekhar commented on src/coverageprocessor.ts:
```diff
@@ -43,6 +43,8 @@ function processResult(result: Result): Result {
   cachedFiles.map((p) => {
     let filename = new Buffer(p.replace(basepath, ''), 'base64').toString('utf8');
+    // add back root part of filename
+    filename = root + filename;
```
Add a short note in the comment about why this is needed and leave a link to #158

kulshekhar commented on src/preprocessor.ts:
```diff
@@ -28,6 +29,9 @@ export function process(src, path, config) {
             });
 
+        // strip root part from path
+        path = path.replace(root, '');
```
- Add a short note in the comment about why this is needed and leave a link to #158 
- using something like 
```
path = path.startsWith(root) ? path.substr(root.length) : path;
```
might be better

4kochi commented:
I does not solve the problem in all cases. But at least it works in our project which have a deeply nested structure. So it might be good enough. 

I will address your comments.

4kochi commented:
I have added some comments and cleaned up the. Not sure though what kind of test I should add. The code coverage test work fine and if there is a problem with the file names they would break. Maybe you can give me a hint.

kulshekhar commented:
I'll see if I can add the tests for this today. If not, I'll at least merge this so that it works for some of you folks at least!
